anvil/src/input_handler.rs

```diff
@@ -394,6 +394,7 @@ impl<BackendData: Backend> AnvilState<BackendData> {
         {
             let mut frame = AxisFrame::new(evt.time_msec()).source(evt.source());
             if horizontal_amount != 0.0 {
+                frame = frame.relative_direction(Axis::Horizontal, evt.relative_direction(Axis::Horizontal));
```

Comment:
So should this now also be done for the vertical code path?

Reply:
Oh, I thought I had that here already. Added.